### PR TITLE
include site_repository check in install and valid

### PR DESCRIPTION
**Pull request by LiNk-NY**
add `site_repository` from env var or option in `valid` and `install`

### CODE REVIEW

**Comment by LiNk-NY** [comment]
@mtmorgan This is ready to go with your approval

File: R/valid.R [hunk 1]
```diff
@@ -112,8 +112,13 @@ valid <-
              site_repository = character())
 {
     stopifnot(
-        is.logical(checkBuilt), length(checkBuilt) == 1L, !is.na(checkBuilt)
+        is.logical(checkBuilt), length(checkBuilt) == 1L, !is.na(checkBuilt),
+        length(site_repository) <= 1L,
+        is.character(site_repository), !any(is.na(site_repository))
```
Comment (mtmorgan) [inline R/valid.R]: Is this `.is_character(site_repository)` (or maybe `length(site_repository) == 0L || .is_scalar_character(site_repository)`

File: R/valid.R [hunk 2]
```diff
@@ -112,8 +112,13 @@ valid <-
...
     )
+    if (!length(site_repository) || !nzchar(site_repository))
+        site_repository <- .repositories_site_repository()
```
Comment (mtmorgan) [inline R/valid.R]: maybe this logic (and the checks on `site_repository`?) belongs in `repsitories_site_repository()`, so just `site_repository <- .repositories_site_repository(site_repositories)` in the code?

**Review by mtmorgan — COMMENTED** [review]
sorry I commented a while ago but somehow don't understand the review process and did not make these public.

File: R/valid.R (hunk 2 above)
Comment (LiNk-NY) [reply R/valid.R]: Thanks for the suggestion! I moved the logic to `.repositories_site_repository(site_repository)`

**Comment by LiNk-NY** [comment]
It's ready for another look. Thanks @mtmorgan

**Review by mtmorgan — APPROVED** [review]

**Pull request closed without merging**